.github/workflows/main.yml

```diff
@@ -8,30 +8,37 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: true
       matrix:
-        plone-version: 
-          - '5.1'
-          - '5.2'
-        python-version: ['2.7']
+        plone-version: [5.2]
+        python-version: [3.6, 3.7, 3.8]
+        os: [ubuntu-latest]
         include:
+          - plone-version: 5.1
+            python-version: 2.7
+            os: ubuntu-18.04
           - plone-version: 5.2
-            python-version: 3.6
-          - plone-version: 5.2
-            python-version: 3.7
-          - plone-version: 5.2
-            python-version: 3.8
+            python-version: 2.7
+            os: ubuntu-18.04
     steps:
       - uses: actions/checkout@v2
-      - name: Set up Python
+      - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
+
+      - name: Install python-dev
+        if: "${{ matrix.python-version == '2.7' }}"
+        run: sudo apt-get install python-dev
+      - name: Install python3-dev
+        if: "${{ matrix.python-version != '2.7' }}"
+        run: sudo apt-get install python3-dev
+
```

Comment:
This is usually not needed, since `actions/setup-python@v2` also installs the dev-headers.
While running output is `python-dev is already the newest version (2.7.15~rc1-1).` or `python3-dev is already the newest version (3.8.2-0ubuntu2).`